src/plugins/intel_cpu/src/nodes/mkldnn_rnn.cpp

```diff
@@ -110,6 +111,59 @@ const std::map<Precision, Precision> MKLDNNRNN::weightsByLayerPrec {
     // {Precision::U8,   Precision::I8},
 };
 
+
+struct RNNKey {
+    std::vector<DnnlBlockedMemoryDescPtr>& inDataDescs;
+    std::vector<DnnlBlockedMemoryDescPtr>& outDataDescs;
+    std::vector<mkldnn::memory::desc>& wDescs;
```

Comment:
Though we store these vectors in the node, it is still dangerous to use references here. The key is copied into the cache, and the vectors lifetime is not equal to the cache lifetime in general case. Though the situation is hypothetical, it is still possible that we can access the references when they are not valid. The pointers are lightweight so copying vectors to with the desc pointers is not very expensive. Regarding `wDescs` I agree that this vector copying is not very cheap, but I think we can sacrifice this overhead for reliability.

Reply:
Done.